app/api/ddi-events-api/base.js

```diff
@@ -1,16 +1,60 @@
 const config = require('../../config')
+
 const wreck = require('@hapi/wreck')
+const { buildPostRequest, addHeaders } = require('../shared')
+const { NotAuthorizedError } = require('../../errors/notAuthorizedError')
+const { ApiErrorFailure } = require('../../errors/apiErrorFailure')
 
 const baseUrl = config.ddiEventsApi.baseUrl
 
-const get = async (endpoint) => {
-  const options = { json: true }
+const get = async (endpoint, user) => {
+  const options = user?.username
+    ? { headers: addHeaders(user) }
+    : {}
 
-  const { payload } = await wreck.get(`${baseUrl}/${endpoint}`, options)
+  const { payload } = await wreck.get(`${baseUrl}/${endpoint}`, { ...options, json: true })
 
   return payload
 }
 
+const post = buildPostRequest(baseUrl)
+
+const callDelete = async (endpoint, actioningUser) => {
```

Comment:
I'd probably just call this 'delete' rather than 'callDelete' (as per other actions get/post/put etc) for consistency but no need to change

Reply:
:( reserved word